### PR TITLE
Clarify environment variable substitution is not recursive

### LINKED ISSUE #3894: [file configuration] Clarify environment variable substitution is not reccursive.
The spec for environment variable substitution, in [file_configuration](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution), should clarify that substitution is not reccursive.

For example:

``` bash
export PING='${PONG}'
export PONG='${PING}'
```

``` yaml
some_yaml_node: ${PING}
```

should evaluate as string `${PONG}`, and not call `getenv()` on `PONG`.

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #3894

## Changes

In the SDK file configuration, clarify environment variable substitution is not recursive.

For non-trivial changes, follow the [change proposal process](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#proposing-a-change).

* [X] Related issues #3894
* [ ] Related [OTEP(s)](https://github.com/open-telemetry/oteps) #
* [ ] Links to the prototypes (when adding or changing features)
* [X] [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md) file updated for non-trivial changes
* [ ] [`spec-compliance-matrix.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) updated if necessary

### CODE REVIEW

**Comment by marcalff** [comment]
cc @jack-berg

File: specification/configuration/file-configuration.md [hunk 1]
```diff
@@ -116,6 +121,7 @@ float_key: ${FLOAT_VALUE}                             # Valid reference to FLOAT
 combo_string_key: foo ${STRING_VALUE} ${FLOAT_VALUE}  # Valid reference to STRING_VALUE and FLOAT_VALUE
 undefined_key: ${UNDEFINED_KEY}                       # Invalid reference, UNDEFINED_KEY is not defined and is replaced with ""
 ${STRING_VALUE}: value                                # Invalid reference, substitution is not valid in mapping keys and reference is ignored
+reccursive_key: ${REPLACE_ME}                         # Valid reference to REPLACE_ME
```
Comment (trask) [inline specification/configuration/file-configuration.md:124]:
```suggestion
recursive_key: ${REPLACE_ME}                          # Valid reference to REPLACE_ME
```

File: specification/configuration/file-configuration.md [hunk 2]
```diff
@@ -131,6 +137,7 @@ float_key: 1.1                              # Interpreted as type float, tag URI
 combo_string_key: foo value 1.1             # Interpreted as type string, tag URI tag:yaml.org,2002:str
 undefined_key:                              # Interpreted as type null, tag URI tag:yaml.org,2002:null
 ${STRING_VALUE}: value                      # Interpreted as type string, tag URI tag:yaml.org,2002:str
+reccursive_key: ${DO_NOT_REPLACE_ME}        # Interpreted as type string, tag URI tag:yaml.org,2002:str
```
Comment (trask) [inline specification/configuration/file-configuration.md:140]:
```suggestion
recursive_key: ${DO_NOT_REPLACE_ME}         # Interpreted as type string, tag URI tag:yaml.org,2002:str
```

File: specification/configuration/file-configuration.md (hunk 1 above)
Comment (reyang) [reply specification/configuration/file-configuration.md:124]: Also in the PR title.

**Review by pellared — APPROVED** [review]
LGTM

I created to issues while reviewing this PR:
- https://github.com/open-telemetry/opentelemetry-specification/issues/3914
- https://github.com/open-telemetry/opentelemetry-specification/issues/3915

File: specification/configuration/file-configuration.md [hunk 3]
```diff
@@ -102,7 +105,9 @@ export STRING_VALUE="value"
 export BOOl_VALUE="true"
 export INT_VALUE="1"
 export FLOAT_VALUE="1.1"
-export INVALID_MAP_VALUE"value\nkey:value" # An invalid attempt to inject a map key into the YAML
+export INVALID_MAP_VALUE="value\nkey:value"           # An invalid attempt to inject a map key into the YAML
+export REPLACE_ME='${DO_NOT_REPLACE_ME}'              # A valid replacement text
+export DO_NOT_REPLACE_ME="Never use this value"       # An invalid attempt to recursively substitute environment variables
```
Comment (reyang) [inline specification/configuration/file-configuration.md:110]: I find the comment confusing. This does not have any recursive reference:

```text
export DO_NOT_REPLACE_ME="Never use this value"
```

Comment (marcalff) [reply specification/configuration/file-configuration.md]: Fixed, removed the confusing comment.

File: specification/configuration/file-configuration.md [hunk 4]
```diff
@@ -102,7 +105,9 @@ export STRING_VALUE="value"
 export BOOl_VALUE="true"
 export INT_VALUE="1"
 export FLOAT_VALUE="1.1"
-export INVALID_MAP_VALUE"value\nkey:value" # An invalid attempt to inject a map key into the YAML
+export INVALID_MAP_VALUE="value\nkey:value"           # An invalid attempt to inject a map key into the YAML
+export REPLACE_ME='${DO_NOT_REPLACE_ME}'              # A valid replacement text
```
Comment (pellared) [inline specification/configuration/file-configuration.md:109]: I think we should add a comment that it is not evaluated to `Never use this value`.

Comment (marcalff) [reply specification/configuration/file-configuration.md]: Fixed.

**Review by pellared — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

**Comment by jack-berg** [comment]
Can you add a changelog entry? Thanks!

**Comment by marcalff** [comment]
> Can you add a changelog entry? Thanks!

Done.

**Comment by jack-berg** [comment]
cc @open-telemetry/configuration-maintainers

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Ping @open-telemetry/configuration-maintainers (otherwise we are good to go)

**Comment by marcalff** [comment]
> Ping @open-telemetry/configuration-maintainers (otherwise we are good to go)

@jack-berg is part of configuration-maitainers, and already approved. Not sure how many approvals are needed.

**Comment by carlosalberto** [comment]
Merging this one.

cc @MrAlias @ocelotl

**Pull request merged**